DigitalLearningSolutions.Data.Tests/TestHelpers/ProgressTestHelper.cs

```diff
@@ -26,5 +36,15 @@ public static class ProgressTestHelper
                 CompleteByDate = completeByDate
             };
         }
+
+        public DateTime? GetSupervisorVerificationRequestedByAspProgressId(int aspProgressId)
+        {
+            return connection.Query<DateTime?>(
+                @"SELECT SupervisorVerificationRequested
+                    FROM aspProgress
+                    WHERE aspProgressId = @aspProgressId",
+                new { aspProgressId }
+            ).Single();

```

Comment:
I assume this must work since the tests pass, but my understanding of the Single() LINQ operator is that it throws if there's not exactly one result, so this can never return null. Can you tell me what I've got wrong here?

Reply:
The column SupervisorVerificationRequested is a nullable datetime, so if the aspProgress record exists but has null it will return null, hence the DateTime? return type.
As you say, this will throw an exception if there is not exactly one result, but since we are looking it up by aspProgressId and using it in a unit test this is the behaviour we would want. i.e. this will only throw an exception if the aspProgress record gets deleted from the database for some reason.